lddwrap/__init__.py

```diff
@@ -137,51 +114,53 @@ def _parse_line(line: str) -> Optional[Dependency]:
+    mem_address = None
     if '=>' in line:
-        if len(parts) != 4:
-            raise RuntimeError(
-                "Expected 4 parts in the line but found {}: {}".format(
-                    len(parts), line))
-
-        soname = None
-        dep_path = None
-        mem_address = None
+        mtch = _LDD_ARROW_OUTPUT_RE.match(line)
+        if not mtch:
+            raise RuntimeError(("Unexpected ldd output. Expect a regex match {}, "
+                                "but got: {!r}").format(_LDD_ARROW_OUTPUT_RE.pattern,
+                                                        line))
         if found:
-            soname = parts[0]
-            if parts[2] != '':
-                dep_path = pathlib.Path(parts[2])
-
-            mem_address = _strip_mem_address(text=parts[3])
+            soname = mtch["soname"]
+            if mtch["dep_path"]:
+                dep_path = pathlib.Path(mtch["dep_path"])
+            if mtch["mem_address"]:
+                mem_address = mtch["mem_address"]
         else:
             if "/" in parts[0]:
-                dep_path = pathlib.Path(parts[0])
+                # This is a special case where the dep_path comes before the arrow
+                # and we have no soname
+                dep_path = pathlib.Path(mtch["soname"])
             else:
-                soname = parts[0]
-
-        return Dependency(
-            soname=soname, path=dep_path, found=found, mem_address=mem_address)
+                soname = mtch["soname"]
     else:
-        if len(parts) != 2:
-            # Please see https://github.com/Parquery/pylddwrap/pull/14
-            if 'no version information available' in line:
-                return None
+        # Please see https://github.com/Parquery/pylddwrap/pull/14
+        if 'no version information available' in line:
+            return None
+
+        mtch = _LDD_NON_ARROW_OUTPUT_RE.match(line)
+        if not mtch:
+            raise RuntimeError(("Unexpected ldd output. Expect a regex match {}, "
+                                "but got: {!r}").format(_LDD_NON_ARROW_OUTPUT_RE.pattern,
+                                                        line))
+        # Special case for linux-vdso
+        if mtch["dep_path"].startswith("linux-vdso"):
+            soname = mtch["dep_path"]
+        else:
+            dep_path = pathlib.Path(mtch["dep_path"])
 
-            raise RuntimeError(
-                "Expected 2 parts in the line but found {}: {}".format(
-                    len(parts), line))
+        found = True
+        mem_address = mtch["mem_address"]
 
-        if parts[0].startswith('linux-vdso'):
-            soname = parts[0]
-            path = None
-        else:
-            soname = None
-            path = pathlib.Path(parts[0])
-
-        return Dependency(
-            soname=soname,
-            path=path,
-            found=True,
-            mem_address=_strip_mem_address(text=parts[1]))
+    # Sanity check to see if it didn't parse garbage
```

Comment:
Could you please add a full-colon (`:`) or a semi-colon (`;`) at the end of the line for better readability? I didn't parse the comment on the first reading due to missing interpunction.